tests/testthat/test-deploymentTarget.R

```diff
@@ -83,46 +197,70 @@ test_that("succeeds if there's a single existing deployment", {
     appName = "test",
     appId = "1",
     username = "ron",
+    account = "ron",
+    server = "example.com",
     version = "999"
   )
-
-  target <- deploymentTarget(app_dir)
-  expect_equal(target$appId, "1")
-  expect_equal(target$username, "ron")
-  expect_equal(target$version, "999")
-
-  target <- deploymentTarget(app_dir, appName = "test")
-  expect_equal(target$appId, "1")
-  expect_equal(target$username, "ron")
+  expect_equal(nrow(deployments(app_dir, accountFilter = "ron", serverFilter = "example.com")), 1)
+  expect_equal(nrow(deployments(app_dir)), 1)
+
+  target <- findDeploymentTarget(app_dir)
+  accountDetails <- target$accountDetails
+  deployment <- target$deployment
+  expect_equal(accountDetails$name, "ron")
+  expect_equal(accountDetails$server, "example.com")
+  expect_equal(deployment$appId, "1")
+  expect_equal(deployment$name, "test")
+  expect_equal(deployment$username, "ron")
+  expect_equal(deployment$account, "ron")
+  expect_equal(deployment$server, "example.com")
+  expect_equal(deployment$version, "999")
+
+  target <- findDeploymentTarget(app_dir, appName = "test")
+  accountDetails <- target$accountDetails
+  deployment <- target$deployment
+  expect_equal(accountDetails$name, "ron")
+  expect_equal(accountDetails$server, "example.com")
+  expect_equal(deployment$appId, "1")
+  expect_equal(deployment$name, "test")
+  expect_equal(deployment$username, "ron")
+  expect_equal(deployment$account, "ron")
+  expect_equal(deployment$server, "example.com")
+  expect_equal(deployment$version, "999")
 })
 
-test_that("errors if single deployment and appId doesn't match", {
+test_that("appId is used even when name does not match", {
   local_temp_config()
```

Comment:
Just occurs to me — do we / should we emit a warning when this happens? Or not worth it?

I guess if it works and does what the user wants, no need to confuse them (especially because of the somewhat confusing difference between `name` and `title`).

Reply:
We do not emit a warning. We talked about emitting warnings earlier in the investigation, but that's because we thought that folks were making mistakes when calling `deployApp()`. In fact, `deployApp()` was making the wrong deployment choices.